### PR TITLE
[Refactor] Change starlet fs URI format

### PR DESCRIPTION
**Pull request by sduzh**
Old URI format: staros://path/to/object?ShardId=shard_id
New URI format: staros://shard_id/path/to/object

The new format make it easier to join paths, e.g, `auto new_string = fmt::format("{}/{}", parent, child)`

## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

### CODE REVIEW

File: be/src/fs/fs_starlet.h [hunk 1]
```diff
@@ -4,11 +4,21 @@
...
-static const char* const kStarletPrefix = "staros://";
+// is_starlet_uri() performs less strict validation than parse_starlet_uri(), which means
+// if is_starlet_uri() returns false parse_starlet_uri() must fail and if is_starlet_uri()
+// returns true parse_starlet_uri() may also fail.
+bool is_starlet_uri(std::string_view uri);
+
+std::string build_starlet_uri(int64_t shard_id, std::string_view path);
+
+// The first element of pair is path, the second element of pair is shard id.
+// If parse_starlet_uri() succeeded, is_starlet_uri() must be true.
```
Comment (kevincai) [inline be/src/fs/fs_starlet.h:20]: an example would be better.

```
//      staros://shardid/over/there
//       \__/    \_____/ \_______/
//        |         |        |     
//      scheme   shard_id   path
```

**Review by kevincai — DISMISSED** [review]

File: be/src/fs/fs_starlet.h (hunk 1 above)
Comment (sduzh) [reply be/src/fs/fs_starlet.h:25]: updated

File: be/src/fs/fs_starlet.cpp [hunk 2]
```diff
@@ -38,23 +39,36 @@ using staros::starlet::fslib::kS3AccessKeySecret;
 using staros::starlet::fslib::kS3OverrideEndpoint;
 using staros::starlet::fslib::kSysRoot;
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/fs/fs_starlet.cpp]: updated

**Review by kevincai — APPROVED** [review]

**Review by tracymacding — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**